worker/buildbot_worker/test/unit/test_runprocess.py

```diff
@@ -857,3 +857,20 @@ def test_statFile_exists(self):
         self.assertEqual(
             st and st[2], 2, "statfile.log exists and size is correct")
         os.remove('statfile.log')
+
+    def test_invalid_utf8(self):
+        # create the log file watcher first
+        rp = self.makeRP()
+        lf = runprocess.LogFileWatcher(rp, 'test', 'invalid_utf8.log',
+                                       follow=False, poll=False)
+        # now write to the log file
+        INVALID_UTF8 = b'\xff'
```

Comment:
Could we check that we don't drop entire log contents and just remove the bad UTF-8? I.e. let's have some valid text here too and then check what we send just after `rp._sendBuffers()`. See `testSendBuffered()` test in `test_runprocess.py` for an example. We'll need to get `FakeWorkerForBuilder` instance out of `makeRP`.